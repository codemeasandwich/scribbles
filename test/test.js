const scribbles = require('../index');
const sample1 = require('./sample1');
const sample2 = require('./sample2');

//=====================================================
//======================================
//=====================================================


describe('.config(...)', () => {
  it('should allow setting global', () => {
    scribbles.config({global:"console"})
    console.log('HelloWorld')
    scribbles.config.reset()
})

describe('.status( .?. )', () => {
  it('+', () => {

  })
})


describe('check defaults', () => {
  it('level output to console', () => {

    const globalDate = Date;
    const mockDate = new Date('14 Oct 1995')
    global.Date = jest.fn().mockImplementation(() => mockDate) // mock Date "new" constructor

    config.levels.forEach(level=>{
      const cl = console[level]
      console[level] = (scribblesOutput)=>{
        expect(scribblesOutput).toMatchSnapshot();
      }
      scribbles[level]('HelloWorld');
      console[level] = cl
    })
    global.Date = globalDate

  })
})
