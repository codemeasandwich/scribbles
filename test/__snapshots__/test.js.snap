// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check defaults level output to console 1`] = `"scribbles:test:master [ ] 1995-10-14T00:00:00.000 #c353530 <error> test.js:25 HelloWorld"`;

exports[`check defaults level output to console 2`] = `"scribbles:test:master [ ] 1995-10-14T00:00:00.000 #c353530 <warn> test.js:25 HelloWorld"`;

exports[`check defaults level output to console 3`] = `"scribbles:test:master [ ] 1995-10-14T00:00:00.000 #c353530 <log> test.js:25 HelloWorld"`;

exports[`check defaults level output to console 4`] = `"scribbles:test:master [ ] 1995-10-14T00:00:00.000 #c353530 <info> test.js:25 HelloWorld"`;

exports[`check defaults level output to console 5`] = `"scribbles:test:master [ ] 1995-10-14T00:00:00.000 #c353530 <debug> test.js:25 HelloWorld"`;
